inc/Engine/Media/PreloadFonts/Frontend/Subscriber.php

```diff
@@ -46,4 +57,34 @@ public static function get_subscribed_events(): array {
 	public function add_preload_fonts_in_head( $items ) {
 		return $this->preload_fonts->add_preload_fonts_in_head( $items );
 	}
+
+	/**
+	 * Gets the list of fonts to be excluded from preloading.
+	 * Merges any existing exclusions with those from the dynamic lists.
+	 *
+	 * @param array $exclusions Array of font URLs to be excluded from preloading.
+	 * @return array
+	 */
+	public function get_exclusions( array $exclusions ): array {
+		$lists = $this->data_manager->get_lists();
+		$lists = isset( $lists->preload_fonts_exclusions ) ? $lists->preload_fonts_exclusions : [];
+
+		// Check that lists is a valid array.
+		if ( ! is_array( $lists ) ) {
+			$lists = [];
+		}
+
+		// Return early if exlusions is empty.
+		if ( empty( $exclusions ) ) {
+			return $lists;
+		}
+
+		// Return early if lists is empty.
+		if ( empty( $lists ) ) {
+			return $exclusions;
+		}
+
+		// Only merge if exclusion and list is not empty.
+		return array_merge( $exclusions, $lists );
+	}
```

Comment:
```suggestion
	public function get_exclusions( array $exclusions ): array {
		$lists = $this->data_manager->get_lists()->preload_fonts_exclusions ?? [];

		// Merge exclusions and lists (handles empty arrays gracefully).
		return array_merge( $exclusions, (array) $lists );
	}
```

What do u think about simplifying it like this?

Reply:
Yes, I thought about this, but it was more of the micro-optimization for me to avoid unnecessary call to merge.

Reply:
I'm OK with both, I proposed this not for optimization but for readability, I consulted copilot now and he thinks that it's better to remove those checks.
Again I'm OK with both solutions just trying to make it easier to maintain this part in the future, but if u see that the new code is not handling all cases plz keep it as it is.